hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegistryEndpointsRefresher.java

```diff
@@ -60,9 +63,20 @@ public synchronized void stop() {
     notifyAll();
   }
 
+  private long getRefreshIntervalMs(boolean firstRefresh) {
+    if (refreshNow) {
+      return minTimeBetweenRefreshesMs;
+    }
+    if (firstRefresh) {
+      return initialDelayMs;
```

Comment:
One minor edge case is initialDelayMs > periodicRefreshMs (like in a misconfiguration), that needs to be handled ?

(or) we can make initialDelayMs = periodicRefreshMs/4 or something like that, essentially a f(periodicRefreshMs) rather than a separate config, whatever works for you, I don't have a preference.

Reply:
I think a new config option is needed, but the default value could be periodicRefreshMs/4 or /5, if users explicitly set it, we will use the value set by users. WDYT?